curriculum/challenges/english/25-front-end-development/workshop-fruit-search-app/67f90574263ebf363f9edd1d.md

```diff
@@ -0,0 +1,255 @@
+---
+id: 67f90574263ebf363f9edd1d
+title: Step 13
+challengeType: 0
+dashedName: step-13
+---
+
+# --description--
+
+Inside the `useEffect` hook, you should set a `700ms` delay before making a request to the Fruit API: `https://fruit-search.freecodecamp.rocks/api/fruits?q=${query}`. After fetching the data, map over the response and update the `results` state with an array of fruit names. Also, make sure to clear the timeout when the component re-renders or unmounts.
```

Comment:
After going through the workshop, this is going to be to much for just one step. Also, there is no mention of error handling in these directions. So I would suggest breaking this step up into multiple. maybe 2 might work. Maybe it needs to be broken up furuther? But just having one step for this is to much.